src/Nethermind/Nethermind.Network/P2P/Subprotocols/Les/LesProtocolHandler.cs

```diff
@@ -0,0 +1,209 @@
+            {
+                throw new SubprotocolException($"No {nameof(StatusMessage)} received prior to communication with {Session.Node:c}.");
+            }
+
+            int size = message.Content.ReadableBytes;
+
+            switch (message.PacketType)
+            {
+                case LesMessageCode.Status:
+                    StatusMessage statusMessage = Deserialize<StatusMessage>(message.Content);
+                    if (NetworkDiagTracer.IsEnabled) NetworkDiagTracer.ReportIncomingMessage(Session.Node.Host, Name, statusMessage.ToString());
+                    Handle(statusMessage);
+                    break;
+            }
+        }
+
+        public void Handle(StatusMessage status)
+        {
+            Metrics.LesStatusesReceived++;
+
+            // set defaults
+            if (!status.AnnounceType.HasValue) status.AnnounceType = 1;
+
+            if (_statusReceived)
+            {
+                throw new SubprotocolException($"{nameof(StatusMessage)} has already been received in the past");
+            }
+
+            _statusReceived = true;
+            if (Logger.IsTrace)
+                Logger.Trace($"LES received status from {Session.Node:c} with" +
+                             Environment.NewLine + $" prot version\t{status.ProtocolVersion}" +
+                             Environment.NewLine + $" network ID\t{status.ChainId}," +
+                             Environment.NewLine + $" genesis hash\t{status.GenesisHash}," +
+                             Environment.NewLine + $" best hash\t{status.BestHash}," +
+                             Environment.NewLine + $" head blockno\t{status.HeadBlockNo}," +
+                             Environment.NewLine + $" difficulty\t{status.TotalDifficulty}" +
+                             Environment.NewLine + $" announce type\t{status.AnnounceType}" +
+                             Environment.NewLine + $" serve headers\t{status.ServeHeaders}" +
+                             Environment.NewLine + $" serve chain since\t{status.ServeChainSince}" +
+                             Environment.NewLine + $" serve recent chain\t{status.ServeRecentChain}" +
+                             Environment.NewLine + $" serve state since\t{status.ServeStateSince}" +
+                             Environment.NewLine + $" serve recent state\t{status.ServeRecentState}" +
+                             Environment.NewLine + $" transaction relay\t{status.TxRelay}" +
+                             Environment.NewLine + $" buffer limit\t{status.BufferLimit}" +
+                             Environment.NewLine + $" max recharge\t{status.MaximumRechargeRate}");
+                             // todo - log max request costs table
+
+            _remoteHeadBlockHash = status.BestHash;
+
+
+            ReceivedProtocolInitMsg(status);
+            LesProtocolInitializedEventArgs eventArgs = new LesProtocolInitializedEventArgs(this)
+            {
```

Comment:
maybe just set a reference to status message in the handler instead of copying? maybe breaks the isolation slightly but also less heavy on memory and easier to maintain